public/app/plugins/datasource/loki/queryUtils.ts

```diff
@@ -241,6 +241,20 @@ export function getLogQueryFromMetricsQuery(query: string): string {
   return `${selector} ${pipelineExpr}`.trim();
 }
 
+export function getLogQueryFromMetricsQueryAtPosition(query: string, position: number): string {
+  if (isLogsQuery(query)) {
+    return query;
+  }
+
+  const metricQuery = getNodesFromQuery(query, [MetricExpr])
+    .reverse() // So we don't get the root metric node
+    .find((node) => node.from <= position && node.to >= position);
```

Comment:
Not sure I understand completely, but I think you can use `findLeaves` from `modifyQuery.ts`?

Reply:
It's because otherwise, the first node that contains the position is the root one:

![imagen](https://github.com/grafana/grafana/assets/1069378/4332c2e3-0f69-4290-8304-b1d44087b8ea)